sdcm/nemesis.py

```diff
@@ -4834,119 +4834,70 @@ def disrupt_bootstrap_streaming_error(self):
 
-        log_follower = new_node.follow_system_log(patterns=[terminate_pattern.log_message])
+        trigger = partial(
+            bootstrapabortmanager.start_bootstrap)
 
-        watcher = partial(
-            self._call_disrupt_func_after_expression_logged,
-            log_follower=log_follower,
-            disrupt_func=new_node.stop_scylla,
-            disrupt_func_kwargs={},
-            delay=0
-        )
+        watcher = partial(bootstrapabortmanager.abort_bootstrap,
+                          log_message=terminate_pattern.log_message,
+                          timeout=terminate_pattern.timeout + instance_start_timeout)
 
-        with EventsSeverityChangerFilter(new_severity=Severity.WARNING,
-                                         event_class=DatabaseLogEvent,
-                                         regex=".*init - Startup failed.*",
-                                         extra_time_to_expiration=30), \
-                adaptive_timeout(operation=Operations.NEW_NODE, node=self.target_node, timeout=3600) as bootstrap_timeout:
-            ParallelObject(objects=[trigger, watcher], timeout=bootstrap_timeout).call_objects(ignore_exceptions=True)
-
-        host_id_searcher = new_node.follow_system_log(patterns=['Setting local host id to'])
-
-        def clean_unbootstrapped_node():
-            new_node_host_ids = []
-            if found_stings := list(host_id_searcher):
-                for line in found_stings:
-                    new_node_host_id = line.split(" ")[-1].strip()
-                    self.log.info("Node %s has host id: %s in log", new_node.name, new_node_host_id)
-                    new_node_host_ids.append(new_node_host_id)
-            self.log.debug("New host was not properly bootstrapped. Terminate it")
-            self._terminate_cluster_node(new_node)
-            self.log.info("Sleep for gossiper updater")
-            time.sleep(60)
-            self.target_node.raft.clean_group0_garbage(raise_exception=True)
-            if new_node_host_ids:
-                for host_id in new_node_host_ids:
-                    self.target_node.run_nodetool(
-                        f"removenode {host_id}", ignore_status=True, retry=3, warning_event_on_exception=True)
-
-            assert self.target_node.raft.is_cluster_topology_consistent(), \
-                "Group0, Token Ring and number of node in cluster are differs. Check logs"
-            self.cluster.check_nodes_up_and_normal()
-            self.log.info("Failed bootstrapped node removed. Cluster is in initial state")
+        with ignore_stream_mutation_fragments_errors(), contextlib.ExitStack() as stack:
+            for expected_start_failed_context in self.target_node.raft.get_severity_change_filters_scylla_start_failed(
+                    terminate_pattern.timeout):
+                stack.enter_context(expected_start_failed_context)
+            try:
+                ParallelObject(objects=[trigger, watcher],
```

Comment:
better to move logic of propper NodeBootStrapAbortHandler configuration and start to rarf/common or even class itself

something like:

bootstrapabortmanager = NodeBootStrapAbortHandler(node=new_node, verification_node=self.target_node)
bootstrapabortmanager.run(abort_log_pattern, abort_action)

  if not new_node.db_up():
      # stop scylla if it was started by scylla-manager-client during setup
       new_node.stop_scylla_server(ignore_status=True, timeout=600)
       # Clean garbage from group 0 and scylla data and restart setup
.............................................................

bootstrapabortmanager.clear()

Reply:
fixed